### ISSUE TITLE
[gardena] Binding shouldn't reach request limit on Husqvarna cloud service

### ISSUE THREAD

**Issue by bodoweiss** [openhab/openhab-addons#10841]
The problem has been initially described in the openHAB community: https://community.openhab.org/t/problems-to-connect-to-gardena-cloud-service-with-the-gardena-binding

## Expected Behavior
The Husqvarna cloud service has an upper limit of 3000 requests a month for one particular API token. When the limit is reached, the operation of the Gardena binding stops. The Gardena binding shouldn't reach the limit of 3000 requests a month. This would assure continous unattended operation.

## Current Behavior
After 20-25 days I have two times in a row been reaching the request limit of the Husquana cloud service by applying the Gardena binding. It started this year after having installed my fifth Gardena device. To restart operation I have to create a new API token manually, that has to be configured in the openHAB administrative interface. This implies manual intervention, that preferrablly should be avoided.

I don't know how to analyze, what causes reaching the limit:
- normal operation, since this binding on my site integrates 5 Gardena devices in openHAB (one mower, three water control valves, one smart sensor)
- error situations, since I find several "Connection lost" and "simultaneous.logins" messages in my logfile

## Possible Solution
It would be helpful to have some control over the number of requests issued by the binding. This could be realized by an additional parameter of the thing "Gardena Smart Home Account", which this binding needs. This parameter could lengthen the interval between two requests and this way would lower the number of requests to the Husqvarna cloud service.

## Steps to Reproduce (for Bugs)
1. Creation of an API token for the Husqvarna cloud service:  https://developer.husqvarnagroup.cloud/ 
2. Installation and configuration of the Gardena binding. It may be necessary to configure 5 Gardena devices (I am aware, that this is unrealistic, but I am willing to test)
3. Wait until the problem occurs (probably longer than 20 days)

## Context
For this binding I very much rely on uninterrupted operation. I use this binding for watering my garden in summer, especially when I am on vacation. When the operation of this binding stops while I am on vacation, my whole garden will dry up and has to be recultivated.

## Your Environment
* Version used: openHAB stable 3.0.2, Gardena binding 3.0.2
* Environment name and version (e.g. Chrome 76, Java 8, Node.js 12.9, ...): OpenJDK Runtime Environment Zulu11.41+75-CA (build 11.0.8+10-LTS)
* Operating System and version (desktop or mobile, Windows 10, Raspbian Buster, ...):  Raspbian GNU/Linux 9.13 (stretch)

**Comment by Rossko57** [comment]
If proposing a user configurable query rate, it may be worth considering to make that dynamic, so that rate could be reduced e.g. at night or other "off peak" periods.

**Comment by Hilbrand** [comment]
Briefly looking at the binding it looks like it restarts the websocket connection each time a refresh command is received by the binding, which triggers an api call. According to the documentation of the gardena binding it's a feature to do this when sending it to `lastUpdate_timestamp` channel. However, the binding does not check if it's send to this channel so it will refresh on any channel it's send to. That is already incorrect. But this is also a misuse of the Refresh command. Because the refresh command is intended for the UI to trigger updating the state of channels. Meaning if you open the UI it could send an update to all channels, and trigger a number of  api calls equal to the number of channels linked at that single ui action. So I would guess fixing this misuse of the refresh command might already fix this problem.

**Comment by Rossko57** [comment]
I don't think UI usually issue REFRESH commands to anything.  However, I do think every Item (and thus every linked channel) gets a REFRESH at system start up. 
It's up to the binding to decide what to do with each one, but clearly making a dozen API calls is the wrong thing to do, eating into your call allowance.

**Comment by lsiepel** [comment]
The binding had some related PR's can you confirm if this is still an issue in 3.4.0?

**Comment by bodoweiss** [comment]
I upgraded to openhab 3.4.2 from version 3.3 just two weeks ago, so these are just first experiences. Also I have connected only 1 of my 5 Gardena devices already, since we have early spring and watering the garden and cutting the lawn isn't necessary at this point of time. Anyway the thing "Gardena Smart Home Account" reports an communication error with the message "Wait until 01:05:58 for an automatic reconnection attempt". This is in an infinite loop. The only workaround I know still helps: I have to create a new API token manually, that has to be configured in the openHAB administrative interface. After that the thing "Gardena Smart Home Account" turns Online.
